app/controllers/application_controller.rb

```diff
@@ -5,6 +5,143 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+
+    enable :sessions unless test?
+    set :session_secret, "secret"
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @user = current_user
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :"tweets/create_tweet"
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets/new' do
+    if !params[:content].empty?
+      @tweet = Tweet.create(content: params[:content], user_id: session[:id])
+      redirect to '/tweets'
+    end
+  end
+
+  get '/users/:username_slug' do
+    @user = User.find_by_slug(params[:username_slug])
+    @tweets = @user.tweets.all
+    erb :'/users/show_tweets'
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'/tweets/edit_tweet'
+    else
+      redirect to '/login'
+    end
   end
 
-end
\ No newline at end of file
+  patch '/tweets/:id' do
+    @tweet = Tweet.find(params[:id])
+    if @tweet.user_id == session[:id] && !params[:content].empty?
+      @tweet.content = params[:content]
+      @tweet.save
+      redirect to '/tweets'
+    elsif @tweet.user_id == session[:id] && params[:content].empty?
```

Comment:
Your `current_user` method could be useful here.

`elsif @tweet.user_id == current_user.id && params[:content].empty?`